cola/widgets/standard.py

```diff
@@ -89,13 +89,26 @@ def export_state(self):
         """Exports data for view save/restore"""
         state = self.windowState()
         maximized = bool(state & Qt.WindowMaximized)
-        return {
+        width, height = self.width(), self.height()
+        if maximized:
+            settings = Settings()
+            settings.load()
+            last_state = settings.get_gui_state(self)
+            if 'width' in last_state and 'height' in last_state:
+                width, height = int(last_state['width']), int(last_state['height'])
+            else:
+                width, height = None, None
+        ret = {
             'x': self.x(),
             'y': self.y(),
-            'width': self.width(),
-            'height': self.height(),
+            'width': width,
+            'height': height,
             'maximized': maximized,
         }
+        if ret['width'] is None or ret['height'] is None:
+            del ret['width']
+            del ret['height']
+        return ret
```

Comment:
Makes sense.  It's unfortunate that we have to read from disk to get the old size.  It'd be nice if we could store the pre-maximized state in memory and then just ask the widgets.

One small sug.. maybe instead of setting them to `None` and then checking afterwards, how about we remove them from the initial `ret = {...}` statement and then do:

```
    if width and height:
        ret['width'] = width
        ret['height'] = height
    return ret
```

at the end so that it avoids adding it in the first place..  Alternatively, is there any downside in not setting them to `None`?  It seems like that'd still have the desired effect since the common code path is that it'll have a previous size.